x/subspaces/types/genesis.go

```diff
@@ -6,227 +6,163 @@ import (
-	return ACLEntry{
-		SubspaceID:  subspaceID,
-		User:        user,
-		Permissions: permissions,
-	}
-}
-
-// Validate returns an error if something is wrong within the entry data
-func (entry ACLEntry) Validate() error {
-	if entry.SubspaceID == 0 {
-		return fmt.Errorf("invalid subspace id: %d", entry.SubspaceID)
-	}
-
-	_, err := sdk.AccAddressFromBech32(entry.User)
-	if err != nil {
-		return fmt.Errorf("invalid user address: %s", entry.User)
-	}
-
-	return nil
-}
-
-// -------------------------------------------------------------------------------------------------------------------
-
-// NewUserGroupMembersEntry returns a new UserGroupMembersEntry instance
-func NewUserGroupMembersEntry(subspaceID uint64, groupID uint32, members []string) UserGroupMembersEntry {
-	return UserGroupMembersEntry{
-		SubspaceID: subspaceID,
-		GroupID:    groupID,
-		Members:    members,
-	}
-}
-
-// Validate returns an error if something is wrong within the entry data
-func (entry UserGroupMembersEntry) Validate() error {
-	if entry.SubspaceID == 0 {
-		return fmt.Errorf("invalid subspace id: %d", entry.SubspaceID)
-	}
-
-	if entry.GroupID == 0 {
-		return fmt.Errorf("invalid group id: %d", entry.GroupID)
-	}
-
-	for _, user := range entry.Members {
-		_, err := sdk.AccAddressFromBech32(user)
-		if err != nil {
-			return fmt.Errorf("invalid user address: %s", user)
-		}
-	}
-
-	return nil
-}
-
-// -------------------------------------------------------------------------------------------------------------------
-
 // NewGenesisState creates a new genesis state
 func NewGenesisState(
-	initialSubspaceID uint64, subspaces []GenesisSubspace, acl []ACLEntry,
-	userGroups []UserGroup, userGroupMembers []UserGroupMembersEntry,
+	initialSubspaceID uint64,
+	subspacesData []SubspaceData,
+	subspaces []Subspace,
+	sections []Section,
+	userPermissions []UserPermission,
+	userGroups []UserGroup,
+	userGroupMembers []UserGroupMemberEntry,
 ) *GenesisState {
 	return &GenesisState{
 		InitialSubspaceID: initialSubspaceID,
+		SubspacesData:     subspacesData,
 		Subspaces:         subspaces,
-		ACL:               acl,
+		Sections:          sections,
+		UserPermissions:   userPermissions,
 		UserGroups:        userGroups,
 		UserGroupsMembers: userGroupMembers,
 	}
 }
 
 // DefaultGenesisState returns a default GenesisState
 func DefaultGenesisState() *GenesisState {
-	return NewGenesisState(1, nil, nil, nil, nil)
+	return NewGenesisState(
+		1,
+		nil,
+		nil,
+		nil,
+		nil,
+		nil,
+		nil,
+	)
 }
 
-// -------------------------------------------------------------------------------------------------------------------
-
 // ValidateGenesis validates the given genesis state and returns an error if something is invalid
 func ValidateGenesis(data *GenesisState) error {
 	// Make sure the initial subspace id is valid
-	if data.InitialSubspaceID <= uint64(len(data.Subspaces)) {
+	if data.InitialSubspaceID == 0 {
```

Comment:
I suggest also to switch from 0 to a `DefaultSubspace` or `RootSubspace` constant

Reply:
I don't agree with naming the subspace id `0` with `DefaultSubspace` or `RootSubspace` since this is an **invalid** value for pretty much any case (except user blocks). Those naming would somehow tell the user that it should be a valid value, when it really should not

Reply:
Ah ok, gotcha. Do we have a `RootSubspace` concept? Like a default one?

Reply:
No, there is not a default or root subspace. Every content will need to be placed inside a subspace that DApp developers need to create beforehand. They will not be able to create content if a non-zero subspace is not used. 

The only subspace that is going to be different is going to be the one with id `0`. This will be used from users to block people they don't want to be bothered by with DTag transfer requests. But it will not allow to create content inside it. 

All the other ones will allow all the operations (content creations, blockage, etc).